src/controllers/media.ts

```diff
@@ -39,57 +90,96 @@ export const getByOsdbHash = async(ctx: Context) => {
     throw err;
   }
 
+  // Fail early if OpenSubtitles reports that it did not recognize the hash
   if (!osMeta.metadata) {
     await FailedLookups.updateOne({ osdbHash }, {}, { upsert: true, setDefaultsOnInsert: true });
     return ctx.body = MESSAGES.notFound;
   }
 
-  const newMetadata = {
-    title: osMeta.metadata.title,
+  let newMetadata = {
+    actors: _.isEmpty(Object.values(osMeta.metadata.cast)) ? null : Object.values(osMeta.metadata.cast),
+    genres: _.isEmpty(osMeta.metadata.genres) ? null : osMeta.metadata.genres,
+    goofs: osMeta.metadata.goofs,
     imdbID: osMeta.metadata.imdbid,
     osdbHash: osMeta.moviehash,
-    year: osMeta.metadata.year,
-    subcount: osMeta.subcount,
-    type: osMeta.type,
-    goofs: osMeta.metadata.goofs,
-    trivia: osMeta.metadata.trivia,
     tagline: osMeta.metadata.tagline,
-    genres: osMeta.metadata.genres,
-    actors: Object.values(osMeta.metadata.cast),
+    title: osMeta.metadata.title,
+    trivia: osMeta.metadata.trivia,
+    type: osMeta.type,
+    year: osMeta.metadata.year,
   };
 
-  // if we're missing values for genres or actors, attempt to hydrate them from imdbAPI instead of OpenSubtitles
-  if ([newMetadata.genres, newMetadata.actors].some(val => val === undefined || []) && newMetadata.imdbID) {
-    try {
-      const imdbData: any = await imdbAPI.get({ id: newMetadata.imdbID });
-      newMetadata.actors = imdbData.actors.split(', ');
-      newMetadata.genres = imdbData.genres.split(', ');
-    } catch (e) {
-      // ignore error, this shouldn't make the request fail
+  try {
+    dbMeta = await MediaMetadata.create(newMetadata);
+    return ctx.body = dbMeta;
+  } catch (e) {
+    if (e.name === 'ValidationError') {
+      // continue for validation errors
```

Comment:
Do we maybe want to annotate this a bit more to explain we use the supplementary API if we're missing certain data